### PR TITLE
[IMP] purchase_discount: fix stock get_price_unit

### PR DESCRIPTION
**Pull request by cubells**
There's the concept of unit price in stock moves that it's used for stock valuation, and this one doesn't have the correct price when using discounts before this commit.

cc @Tecnativa

### CODE REVIEW

File: purchase_discount/models/purchase_order.py [hunk 1]
```diff
@@ -28,3 +28,24 @@ def _compute_amount(self):
         ('discount_limit', 'CHECK (discount <= 100.0)',
          'Discount must be lower than 100%.'),
     ]
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.model
+    def _prepare_inv_line(self, account_id, order_line):
```
Comment (pedrobaeza) [inline purchase_discount/models/purchase_order.py]: This method is not needed on v9

File: purchase_discount/__openerp__.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
               "Acysos S.L., "
               "Tecnativa, "
               "ACSONE SA/NV,"
+              "AvanzOSC, S.L., "
               "Odoo Community Association (OCA)",
     "version": "9.0.1.0.1",
```
Comment (pedrobaeza) [inline purchase_discount/__openerp__.py]: Change version number

File: purchase_discount/__openerp__.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
               "Acysos S.L., "
               "Tecnativa, "
               "ACSONE SA/NV,"
+              "AvanzOSC, S.L., "
```
Comment (pedrobaeza) [inline purchase_discount/__openerp__.py]: This little patch shouldn't be included in authors.

File: purchase_discount/models/purchase_order.py [hunk 4]
```diff
@@ -28,3 +28,17 @@ def _compute_amount(self):
...
+        value before calling super and restoring it later for assuring
+        maximum inheritability.
+        """
+        if self.discount:
+            price_unit = self.price_unit
+            self.price_unit *= (100 - self.discount) / 100
+        price = super(PurchaseOrderLine, self)._get_stock_move_price_unit()
+        if self.discount:
+            self.price_unit = price_unit
+        return price
```
Comment (sergio-teruel) [inline purchase_discount/models/purchase_order.py:44]: Is multi?? Maybe you're missing a loop...

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: purchase_discount/models/purchase_order.py (hunk 4 above)
Comment (pedrobaeza) [reply purchase_discount/models/purchase_order.py:44]: No, it isn't. super method has self.ensure_one()

Comment (sergio-teruel) [reply purchase_discount/models/purchase_order.py:44]: Oks..

**Review by sergio-teruel — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**